### ISSUE TITLE
Firefox tab crashes by loading bigger studies (> 2000 files)

### ISSUE THREAD

**Issue by kursawero** [ddRPB/rpb-uploader#47]
We need to optimize the memory consumption on reading through all files that are dropped into the UI upload element

**Comment by kursawero** [comment]
Reproduced some crashes - all with "out of memory". Behavior depends on side conditions, like: other tabs that are open in parallel. Restarting Firefox, for instance, already helps and a data set that could not load before passes.

Found another issue. Non RT related modality files were not parsed for the "SOPInstanceUID", but the series view was organized based on that. Not sure if such caused additional memory consumption.

Fixed version with fresh started Firefox was able to read ~4000 file ~ 4GB. Upload of  ~2000 files worked.

**Comment by kursawero** [comment]
Seems that the application itself is stable, but the process takes too much time ~2h for 4 GB. Will merge current state and follow up with a new issue. -> https://github.com/ddRPB/rpb-uploader/issues/48
